images/handlers.go

```diff
@@ -182,35 +182,41 @@ func SetChildrenLabels(manager content.Manager, f HandlerFunc) HandlerFunc {
 	}
 }
 
-// FilterPlatform is a handler wrapper which limits the descriptors returned
-// by a handler to a single platform.
-func FilterPlatform(platform string, f HandlerFunc) HandlerFunc {
+// FilterPlatforms is a handler wrapper which limits the descriptors returned
+// by a handler to the specified platforms.
+func FilterPlatforms(f HandlerFunc, platformList ...string) HandlerFunc {
 	return func(ctx context.Context, desc ocispec.Descriptor) ([]ocispec.Descriptor, error) {
 		children, err := f(ctx, desc)
 		if err != nil {
 			return children, err
 		}
 
 		var descs []ocispec.Descriptor
-		if platform != "" && isMultiPlatform(desc.MediaType) {
-			p, err := platforms.Parse(platform)
-			if err != nil {
-				return nil, err
-			}
-			matcher := platforms.NewMatcher(p)
 
-			for _, d := range children {
-				if d.Platform == nil || matcher.Match(*d.Platform) {
-					descs = append(descs, d)
+		if len(platformList) == 0 {
+			descs = children
+		} else {
+			for _, platform := range platformList {
+				if platform != "" && isMultiPlatform(desc.MediaType) {
```

Comment:
The `isMultiPlatform(desc.MediaType)` could be joined with the length check

Reply:
Also not sure `platform == ""` is valid, so can probably just remove that since empty list would replace that condition

Reply:
I don't know about returning `descs = children` if the image is not multi-platform anyway--if I ask for a specific platform or platforms, and the image isn't multi-platform should there be an error case or some other (knowingly imperfect) matching on the image config OS/arch string(s)?

I do think the inner `if` can probably ignore empty platform strings as `platforms.Parse()` should handle empty strings and error out if necessary.